File: code/sorting/selection_sort/selection_sort.swift
/* Part of Cosmos by OpenGenus Foundation */

//
//  selection_sort.swift
//  Created by DaiPei on 2017/10/11.
//

import Foundation

func selectionSort(_ array: inout [Int]) {
    for i in stride(from: 0, to: array.count - 1, by: 1) {
        var min = i
        for j in i+1..<array.count {
            if array[j] < array[min] {
                min = j
            }
        }
        swap(&array, at: min, and: i)
    }
}

private func swap(_ array: inout [Int], at indexA: Int, and indexB: Int) {
    let tmp = array[indexA]
    array[indexA] = array[indexB]
    array[indexB] = tmp
}

File: code/sorting/src/selection_sort/selection_sort.rs
// Part of Cosmos by OpenGenus Foundation
fn selection_sort(mut arr :Vec<i32>) -> Vec<i32> {
    let len = arr.len();
    
    for i in 0..len-1 {
        let mut min = i;
        for j in i+1..len {
            if arr[j] < arr[min] {
                min  = j;
            }
        }
        if min != i{
            arr.swap(i, min);
        }
    }

    arr
}

fn main() {
    let arr = vec![1, 3, 43, 2, 14, 53, 12, 87, 33];
    let arr = selection_sort(arr);
    println!("Sorted array: {:?}", arr);
}